Sabnzbd config flow

shaiu opened the pull request:
## Breaking change
The Sabnzbd integration has been migrated to configuration via the UI. Configuring Sabnzbd via YAML configuration has now been deprecated and will be removed in a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this release; and thus can be safely removed from your YAML configuration after upgrading.

## Proposed change
Add support for config flow for Sabnzbd component
Adds all sensors by default


## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure


[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

shaiu committed:
Add config flow for Sabnzbd component

shaiu committed:
back to upstream

shaiu committed:
handle import yaml configuration

homeassistant commented:
Hello @shaiu,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`86bb287f642fa926292fa5c915e103e35da731a1`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;core&#x2F;pull&#x2F;68138&#x2F;commits&#x2F;86bb287f642fa926292fa5c915e103e35da731a1) - This commit has something that looks like an email address (shai.ungar@cyren.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_data):
```
You forgot the `async_setup` method to import existing YAML config.

shaiu replied:
I'm not sure exactly what you mean here. I saw all examples do `async_step_import` in order to import existing YAML config. What am I missing?

balloob replied:
you need to trigger that from `async_setup`

shaiu replied:
done

balloob commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -1,234 +1,58 @@
-    """Check if we can reach SABnzbd."""
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the SabNzbd Component."""
+    sab_api = await get_client(hass, entry.data)
```
This should raise `ConfigEntryNotReady` errors if it can't connect to the host, and if authentication fails, it should log an error and `return False`.

shaiu replied:
done

balloob commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -1,234 +1,58 @@
     {vol.Optional(ATTR_SPEED, default=DEFAULT_SPEED_LIMIT): cv.string}
 )
 
-CONFIG_SCHEMA = vol.Schema(
```
Keep this and import it into a config entry.

shaiu replied:
I'm not sure how to do this

balloob replied:
Search how `SOURCE_IMPORT` is used in `__init__.py` together with `async_step_import` in `config_flow.py` in other integrations

shaiu replied:
I moved the vol.Schema to `config_flow.py` under the name `USER_SCHEMA`

balloob replied:
You need ot keep `CONFIG_SCHEMA`. It's a special name that Home Assistant looks up and makes sure it's correct before passing it to `async_setup`.

shaiu replied:
Done

balloob commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -1,234 +1,58 @@
 
 @callback
-def async_setup_sabnzbd(hass, sab_api, config, name):
+def async_setup_sabnzbd(hass, sab_api):
```
Why a standalone method ?

shaiu replied:
not sure what you mean

balloob replied:
it's only used once, why not make it part of async_setup_entry

shaiu replied:
done

balloob commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -254,7 +78,6 @@ async def async_service_handler(service: ServiceCall) -> None:
 
     async def async_update_sabnzbd(now):
```
Why does this service exist? We already have a `homeassistant.update_entity` service

shaiu replied:
this is in order to update stats from sabnzbd every UPDATE_INTERVAL (timedelta(seconds=30))

balloob replied:
Oops, thought this was a service call but it's for time. This is ok.

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+            await get_client(self.hass, user_input)
+
+        except AuthenticationError:
+            errors["base"] = "cannot_connect"
```
Authentication error is not the same as cannot connect. One is an incorrect host, the other an incorrect api key.

shaiu replied:
done

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+        vol.Optional(CONF_PATH): str,
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
```
It would be easier to just request the URL ?

shaiu replied:
done

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
```
Why request a name, can't we get name from the server ?

shaiu replied:
sabnzbd client doesn't expose that.

balloob commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,114 @@
+    SabnzbdSensorEntityDescription(
+        key="current_status",
+        name="Status",
+        field_name="status",
```
What is the difference between `key` and `field_name`  ?

shaiu replied:
`field_name` is internal to sabnzbd api and `key` is for the component

balloob replied:
Why not align them?

shaiu replied:
I can't change the name in `field_name` as it's an API of sabnzbd. Some of the `field name` have unreadable names that are not user friendly so hence the `key` field.

balloob replied:
But the `name` field is already there to be readable? The entity ID is also based on the `name` field.

shaiu replied:
right. Aligned them.
Thanks.

balloob commented on homeassistant/components/sabnzbd/strings.json:
```diff
@@ -0,0 +1,25 @@
+    "step": {
+      "user": {
+        "title": "Sabnzbd",
+        "description": "If you need help with the configuration have a look here: https://www.home-assistant.io/integrations/sabnzbd/",
```
Documentation is now linked in config flow header. This is not needed.
```suggestion
```

balloob commented on homeassistant/components/sabnzbd/strings.json:
```diff
@@ -0,0 +1,25 @@
+  "config": {
+    "step": {
+      "user": {
+        "title": "Sabnzbd",
```
Titles that are the same as manifest will automatically be added. This is not needed.
```suggestion
```

balloob commented on homeassistant/components/sabnzbd/strings.json:
```diff
@@ -0,0 +1,25 @@
+      "invalid_api_key": "[%key:common::config_flow::error::invalid_api_key%]"
+    },
+    "abort": {
+      "single_instance_allowed": "[%key:common::config_flow::abort::single_instance_allowed%]"
```
Why is only a single instance allowed? I also didn't notice you using this error.

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
```
Why?

shaiu replied:
right. removed

balloob commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,80 @@
+            errors = await self._async_validate_input(user_input)
+
+            if not errors:
+                return self.async_create_entry(
```
Would you be able to set a unique ID? Does the server have a unique ID? Otherwise the API key would be unique I guess.

shaiu replied:
I don't see one. The `CONF_NAME` is sort of a unique ID

balloob replied:
Names are not unique as they are user configurable.

shaiu replied:
right. so using now the API_KEY (first 12 chars) as unique ID

shaiu commented:
@balloob Thanks for the review. I'll work on the comments

shaiu committed:
Update homeassistant/components/sabnzbd/strings.json

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

shaiu committed:
Update homeassistant/components/sabnzbd/strings.json

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

shaiu committed:
go over comments in code review

shaiu commented:
@balloob I went over the comments.
Some of them I fixed and I need guidance in some of them.
Thanks

shaiu committed:
1. call import from async_setup
2. use API KEY as unique ID for the entry title

shaiu committed:
fix tests

shaiu commented:
@balloob went over comments ad fixed everything.
Thanks.

shaiu commented:
@balloob ping

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#68138).

Once you do that we will be able to review and accept this pull request.

Thanks!

shaiu committed:
align key and field_name

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#68138).

Once you do that we will be able to review and accept this pull request.

Thanks!

shaiu commented:
@balloob fixed last comment
Thanks

shaiu commented:
> Hi @shaiu, Your last push contains merge instead of rebase. Please undo it and correctly rebase your branch. It now contains 104 commits in 554 changed files.
> 
> Thanks

Yes I just saw it. fixing

shaiu commented:
@thecode fixed CI issues. You can run again

balloob commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -1,319 +1,119 @@
-    )
+        async_track_time_interval(hass, async_update_sabnzbd, UPDATE_INTERVAL)
+
+    async_setup_sabnzbd(hass, sab_api)
```
There is no need this needs to be a separate method.

shaiu replied:
removed method

balloob commented on homeassistant/components/sabnzbd/const.py:
```diff
@@ -0,0 +1,26 @@
+ATTR_SPEED = "speed"
+BASE_URL_FORMAT = "{}://{}:{}/"
+CONFIG_FILE = "sabnzbd.conf"
+DEFAULT_URL = "localhost:8080"
```
This is a bad default URL. Most people run Home Assistant on a dedicated device and this probably won't work. Better to just leave it blank.

shaiu replied:
removed the default and marked URL as Required

balloob commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,103 @@
+        key="noofslots_total",
+        name="Queue Count",
+    ),
+    SabnzbdSensorEntityDescription(
+        key="day_size",
+        name="Daily Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="week_size",
+        name="Weekly Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="month_size",
+        name="Monthly Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="total_size",
+        name="Total",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
```
It is overkill to have the same data basically enabled 4 times. Let's disable 3 by default by adding `entity_registry_enabled_default=False,`.

You can also add `state_class=SensorStateClass.TOTAL_INCREASING,` so statistics will be generated for these sensors.

shaiu replied:
added SensorStateClass to all sensors

balloob commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,103 @@
+    ),
+    SabnzbdSensorEntityDescription(
+        key="noofslots_total",
+        name="Queue Count",
```
This is a total
```suggestion
        name="Queue Count",
        state_class=SensorStateClass.TOTAL,
```

balloob replied:
All other sensors also fit one of these 3 sensor classes. See the docs.

shaiu replied:
added SensorStateClass to all sensors

balloob commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,103 @@
+    SabnzbdSensorEntityDescription(
+        key="kbpersec",
+        name="Speed",
+        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
```
```suggestion
        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
        state_class=SensorStateClass.MEASUREMENT,
```

shaiu replied:
added SensorStateClass to all sensors

shaiu committed:
1. add SensorStateClass to sensors
2. remove default URL and make it Required
3. update strings.json
4. bring back CONFIG_SCHEMA

shaiu commented:
@balloob Thanks a lot for the comments.
I have fixed them all.

balloob commented:
Nice! 🎉 🐬

MartinHjelmare commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -147,88 +39,50 @@ class SabnzbdSensorEntityDescription(SensorEntityDescription, SabnzbdRequiredKey
+                vol.Required(CONF_API_KEY): str,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Required(CONF_URL): str,
+                vol.Optional(CONF_PATH): str,
```
We should not remove the legacy options completely from the config schema, but mark them as deprecated instead with `cv.deprecated`. This will avoid a breaking change for users.

MartinHjelmare commented on homeassistant/components/sabnzbd/config_flow.py:
```diff
@@ -0,0 +1,79 @@
+        """Import sabnzbd config from configuration.yaml."""
+        import_data[CONF_URL] = (
+            ("https://" if import_data[CONF_SSL] else "http://")
+            + import_data[CONF_HOST]
```
Please use an f-string instead, split over multiple lines. We don't need to use `+` to concatate string parts.

MartinHjelmare commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -147,88 +39,50 @@ class SabnzbdSensorEntityDescription(SensorEntityDescription, SabnzbdRequiredKey
+
+    sab_api_data = SabnzbdApiData(sab_api)
 
     async def async_service_handler(service: ServiceCall) -> None:
```
If we allow more than one config entry, we need to identify which config entry, ie which api instance, is targeted during service calls, and also not register the services if they are already registered.

Preferably also unregister the services when the last loaded config entry is unloaded.

MartinHjelmare replied:
Here's an example of how the Simplisafe integration does this. How to identify the api depends on what identifiers are available.

https://github.com/home-assistant/core/blob/21f9b19fad8c61db87e4ece97684a48a287ce4d8/homeassistant/components/simplisafe/__init__.py#L330-L394

https://github.com/home-assistant/core/blob/21f9b19fad8c61db87e4ece97684a48a287ce4d8/homeassistant/components/simplisafe/__init__.py#L426-L435

MartinHjelmare commented on homeassistant/components/sabnzbd/const.py:
```diff
@@ -0,0 +1,25 @@
+DATA_SABNZBD = "sabnzbd"
+
+ATTR_SPEED = "speed"
+BASE_URL_FORMAT = "{}://{}:{}/"
```
`BASE_URL_FORMAT` is not used anymore.

MartinHjelmare commented on homeassistant/components/sabnzbd/const.py:
```diff
@@ -0,0 +1,25 @@
+
+ATTR_SPEED = "speed"
+BASE_URL_FORMAT = "{}://{}:{}/"
+CONFIG_FILE = "sabnzbd.conf"
```
As above.

MartinHjelmare commented on homeassistant/components/sabnzbd/const.py:
```diff
@@ -0,0 +1,25 @@
+from datetime import timedelta
+
+DOMAIN = "sabnzbd"
+DATA_SABNZBD = "sabnzbd"
```
Not used.

MartinHjelmare commented on homeassistant/components/sabnzbd/sab.py:
```diff
@@ -0,0 +1,27 @@
+        await sab_api.check_available()
+    except SabnzbdApiException as exception:
+        _LOGGER.error("Connection to SABnzbd API failed: %s", exception.message)
+        return False
```
Inconsistent return value. We should return None instead but I'd recommend raising an exception instead and letting the caller catch that exception. It will make the return value more strict.

MartinHjelmare commented on homeassistant/components/sabnzbd/sab.py:
```diff
@@ -0,0 +1,27 @@
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+
+async def get_client(hass: HomeAssistant, data):
```
Please add type annotations for the whole signature if adding any type annotations for a function.

MartinHjelmare commented on homeassistant/components/sabnzbd/sab.py:
```diff
@@ -0,0 +1,27 @@
+    try:
+        await sab_api.check_available()
+    except SabnzbdApiException as exception:
+        _LOGGER.error("Connection to SABnzbd API failed: %s", exception.message)
```
Don't log an error if we raise `ConfigEntryNotReady`. Include the message as an exception argument instead. Otherwise we'll spam the log with the same error message when retrying config entry setup.

MartinHjelmare commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,120 @@
-    SabnzbdSensorEntityDescription,
+
+from . import DOMAIN, SIGNAL_SABNZBD_UPDATED, SabnzbdApiData
+from ...config_entries import ConfigEntry
```
Please don't use relative imports for extra package imports. It's not compatible if this integration would be copied to a custom integration.

Use absolute imports for non package names.

MartinHjelmare commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,120 @@
+from ...const import DATA_GIGABYTES, DATA_MEGABYTES, DATA_RATE_MEGABYTES_PER_SECOND
+from ...core import HomeAssistant
+from ...helpers.entity_platform import AddEntitiesCallback
+from .const import KEY_API, KEY_NAME
```
Relative imports for intra package imports is good!

MartinHjelmare commented on homeassistant/components/sabnzbd/sensor.py:
```diff
@@ -1,39 +1,120 @@
+
+    sab_api = hass.data[DOMAIN][config_entry.entry_id][KEY_API]
+    client_name = hass.data[DOMAIN][config_entry.entry_id][KEY_NAME]
+    sab_api_data = SabnzbdApiData(sab_api)
```
We already create an instance of `SabnzbdApiData` during config entry setup in the init module. We shouldn't need more than one instance of that class.

MartinHjelmare commented on homeassistant/components/sabnzbd/strings.json:
```diff
@@ -0,0 +1,18 @@
+    },
+    "error": {
+      "cannot_connect": "[%key:common::config_flow::error::cannot_connect%]",
+      "invalid_api_key": "[%key:common::config_flow::error::invalid_api_key%]"
```
`invalid_api_key` error is never used. If the api key can be invalid we should check for that during config flow validation.

MartinHjelmare commented on homeassistant/components/sabnzbd/errors.py:
```diff
@@ -0,0 +1,10 @@
+from homeassistant.exceptions import HomeAssistantError
+
+
+class AuthenticationError(HomeAssistantError):
```
These exceptions are never used.

MartinHjelmare commented on tests/components/sabnzbd/test_config_flow.py:
```diff
@@ -0,0 +1,113 @@
+        "homeassistant.components.sabnzbd.sab.SabnzbdApi.check_available",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
```
Please patch the integration setup function for all create entry results to avoid setting up the whole integration. See the scaffold test template for example.

MartinHjelmare commented on tests/components/sabnzbd/test_config_flow.py:
```diff
@@ -0,0 +1,113 @@
+        "homeassistant.components.sabnzbd.sab.SabnzbdApi.check_available",
+        return_value=True,
+    ):
+        MockConfigEntry(
```
We can move adding the mock config entry up outside the patch context manager.

MartinHjelmare commented on tests/components/sabnzbd/test_config_flow.py:
```diff
@@ -0,0 +1,113 @@
+
+
+async def test_integration_already_exists(hass):
+    """Test we only allow a single config flow."""
```
We allow more than one config entry so not sure what this test is aiming at testing.

MartinHjelmare commented on homeassistant/components/sabnzbd/__init__.py:
```diff
@@ -254,7 +108,6 @@ async def async_service_handler(service: ServiceCall) -> None:
-
         try:
             await sab_api.refresh_data()
             async_dispatcher_send(hass, SIGNAL_SABNZBD_UPDATED, None)
```
Side note: A future improvement would be to use our DataUpdateCoordinator to handle refreshing data.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

MartinHjelmare reviewed (commented):
Please address the comments in a new PR. Thanks!

MartinHjelmare commented:
Please add a breaking change paragraph in the PR description. See old PRs that introduced config flow and imported the config YAML for a template paragraph.

frenck commented:
This PR has been merged without documentation, please open up a PR to adjust our end-user documentation.

shaiu commented:
> Please add a breaking change paragraph in the PR description. See old PRs that introduced config flow and imported the config YAML for a template paragraph.

Done

shaiu commented:
> Please address the comments in a new PR. Thanks!

https://github.com/home-assistant/core/pull/70981

shaiu commented:
@frenck I'm working now on the documentation
@MartinHjelmare and @frenck I just want
